tests/unit/no-top-level-side-effects.test.ts

```diff
@@ -1391,6 +1391,52 @@ const invalid: RuleTester.InvalidTestCase[] = [
           endColumn: 31
         }
       ]
+    },
+    {
+      code: `const x = foo(bar());`,
+      options: [{allowedCalls: ['foo']}],
+      errors: [
+        {
+          messageId: '0',
+          line: 1,
+          column: 15,
+          endLine: 1,
+          endColumn: 20
+        }
+      ]
+    },
+    {
+      code: `const x = foo(bar());`,
+      options: [{allowedCalls: ['bar']}],
+      errors: [
+        {
+          messageId: '0',
+          line: 1,
+          column: 11,
+          endLine: 1,
+          endColumn: 21
+        }
+      ]
+    },
+    {
+      code: `const x = foo(bar());`,
```

Comment:
Should test the same for `new Foo(new Bar())` as well as mixes of function calls and constructor calls (e.g. `new Foo(bar())`).

Reply:
Resolved with 250ded3819abac82b1034f0c88310d5932086db5